src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/MagicWorkbench.java

```diff
@@ -68,9 +68,10 @@ public void onInteract(Player p, Block b) {
         }
     }
 
-    private void craft(Inventory inv, Block dispenser, Player p, Block b, ItemStack output) {
+    @ParametersAreNonnullByDefault
+    private void craft(Inventory inv, Dispenser dispenser, Player p, Block b, ItemStack output) {
         Inventory fakeInv = createVirtualInventory(inv);
-        Inventory outputInv = findOutputInventory(output, dispenser, inv, fakeInv);
+        Inventory outputInv = findOutputInventory(output, dispenser.getBlock(), inv, fakeInv);
```

Comment:
Same here, `#getBlock()` is a bit expensive and completely avoidable by just passing the `Block` object via a parameter